src/main/java/net/snowflake/ingest/utils/SubColumnFinder.java

```diff
@@ -15,55 +14,67 @@
 import org.apache.parquet.schema.MessageType;
 import org.apache.parquet.schema.Type;
 
-/** Helper class to find all leaf sub-columns in an immutable schema given a dot path. */
+/** Helper class to find all leaf sub-columns in an immutable schema given a fieldId. */
 public class SubColumnFinder {
-  static class SubtreeInterval {
+  static class SubtreeInfo {
     final int startTag;
     final int endTag;
+    final String dotPath;
 
-    SubtreeInterval(int startTag, int endTag) {
+    SubtreeInfo(int startTag, int endTag, String dotPath) {
       this.startTag = startTag;
       this.endTag = endTag;
+      this.dotPath = dotPath;
     }
   }
 
   private final List<String> list;
```

Comment:
add code comments on what is in list, what is in accessMap, similarly for startTag/endTag/SubtreeInfo/all public methods?

Reply:
Added